### ISSUE TITLE
User notifications on PAF and contract transition in compliance flow

### ISSUE THREAD

**Issue by fourthletter** [HyphaApp/hypha#3066]
## Summary

OTF needs to inform the users involved in the process of PAF and contract transition through the webapp.  This is so they can be kept up-to-date of the PAF and contract status, and be notified if they are required to do anything.

## Users involved
- Contractor: reviews and signs the contract

- Programme Manager Staff: they are the contractor's point of contact within OTF.  They submit the PAF to Compliance for approvals.

- Approver + Staff user group: they are the contracting user group that reviews the PAF and to check it has the required information.
- 
- Compliance Attorney: they are the contracting user group that reviews the PAF and to check it has the required information.

- Finance: they are the finance staff that reviews the budget is correct.

## Notification method

It's been decided that 1) OTF Staff will receive notifications via Slack and email, and 2) contractors will receive notifications via the email address defined in their account. 

## Communication style

The current OTF-to-contractor communication style is to provide the minimum level of detail in the email and to require them to log-in to read the full message.

## Notification Flow

PM submits Scope and PAF
- [ ] Compliance notified to approve

Finance approves PAF
- [ ] Compliance notified once Finance approves

Compliance approves PAF
- [ ] CEO notified once Compliance approves

CEO approves PAF
- [ ] Compliance notified, begins drafting contract
- [ ] PM notified of approved PAF

Compliance uploads unsigned contract
- [ ] PM notified to send to Contractor via webapp
- [ ] Or, does Compliance send directly to Contractor point of contact via webapp?
- [ ] If so, notify PM that contract is sent to contractor

Contractor returns signed contract or requests revisions via webapp
- [ ] Compliance notified
- [ ] PM notified

If Contractor requests revisions:
- [ ] Compliance revises, reuploads and begins again at step 5.

When Contractor signs:
- [ ] Compliance obtains CEO approval (signature via adobesign) and uploads fully executed document
- [ ] Finance notified
- [ ] Begins intaact process
- [ ] PM notified
- [ ] Returns fully executed contract to Contractor
- [ ] Compliance notified confirming upload submission is successful???

Total Notification Count:
PM:  4
CEO:  2 
Finance: 2
Compliance:  3 or 4

**Comment by frjo** [comment]
- [x] We should rename "/apply/projects/4/simplified/" to something like "/apply/projects/4/approval/". Also rename "ProjectDetailSimplifiedView".
- [x] The link to the uploads contract in the "Contracts" box should open in a new tab, "target _blank"
- [x] When an applicant uploads a signed contract an e-mail is sent to the applicant but not to the "Compliance Email" address from project settings. I do not think we need the mail to the applicant but need to add one to compliance.
- [x] When a user with roles "contracting + approver" tries to approves a contract they get a 404. Works for admins.
- [x] When a user has approved a PAF I think the PAF should be removed from that users dashboard.
- [ ] Should "Approve contract" button be moved up to the "Action to take" box?
- [x] E-mail should be sent out to "Compliance Email" address when all "PAF Reviewers Roles" have approved a PAF. The project should also show in a "Final approving requeried" table on the Dashboard for Contracting+Approver users.
- [ ] Do we need to separate "Compliance Email" setting to one address for finance users and one for contracting users? Or should we send directly to address is user accounts that has the role in question?
- [x] Fix misspelling of "venor", should be "vendor"

**Comment by frjo** [comment]
@fourthletter Are not all the roles ment to look at and approve the PAF in parallel?

Hypha has no idea who is CEO etc.

**Comment by fourthletter** [comment]
> @fourthletter Are not all the roles ment to look at and approve the PAF in parallel?
> 
> Hypha has no idea who is CEO etc.

BT created an approval workflow that helps to each user role and WHEN they review and approve the PAF: https://miro.com/app/board/uXjVPNCmjCQ=/

**Comment by frjo** [comment]
We have these user roles, what actual people do we expect having these roles?

* Staff + approver -> ?
* Finance -> ?
* Finance + approver -> ?
* Contracting -> ?
* Contracting + approver -> ?

**Comment by fourthletter** [comment]
> We have these user roles, what actual people do we expect having these roles?
> 
> * Staff + approver -> ?
> * Finance -> ?
> * Finance + approver -> ?
> * Contracting -> ?
> * Contracting + approver -> ?

Thank you for flagging this. We will create a similar approval workflow for each of the **user group** on Hypha as opposed to only official staff titles within our oganization.

**Comment by fourthletter** [comment]
On January 10 standup we discussed these points. Please let me know if I captured any main points or decisions incorrectly:

> * [ ]  When a user with roles "contracting + approver" tries to approves a contract they get a 404. Works for admins.

The staff user group is the one to finalize the contract or trigger invoicing on Hypha. We will remove this button for other user groups. 

> * [ ]  When a user has approved a PAF I think the PAF should be removed from that users dashboard.

This makes sense. A PAF is not like an application, but is it an invoice? Do we need to have a record?

> * [ ]  Should "Approve contract" button be moved up to the "Action to take" box?

We are keeping the "Approve contract" button there for now, but will consider distinct button colors in the future

> * [ ]  Do we need to separate "Compliance Email" setting to one address for finance users and one for contracting users? Or should we send directly to address is user accounts that has the role in question?

We decided email notifications could go to a 'group email' and default to individual emails if group email is not provided

**Comment by fourthletter** [comment]
Thank you! 💯 

Users will be able to add 'user group emails' to the projects page via Wagtail Admin > Projects > Project Settings

![image](https://user-images.githubusercontent.com/20019656/215491942-98c21558-98d3-464b-901b-ae2e8e83f442.png)

**Comment by frjo** [comment]
@fourthletter @sandeepsajan0 Now after that PR #3102 is done, what is left in the issue?

**Comment by fourthletter** [comment]
I think @sandeepsajan0 did a great job capturing all the notifications overall. We could close this issue. 

I still need to test for the complete user journey for the **contracting + approver role**, this is the FINAL approver that has their own dashboard and notifications.

**Comment by sandeepsajan0** [comment]
I think we have covered most of the things, we can have another issue if we notice other things to be changed.
